src/client/common/errors/types.ts

```diff
@@ -10,6 +11,17 @@ export abstract class BaseError extends Error {
     }
 }
 
+export abstract class BaseKernelError extends Error {
```

Comment:
All kernel errors will have the `kernelConnectionMetadata`
Ideally we should include the `resource`, but don't need that just yet, hence not adding it now.